src/metatensor/models/utils/omegaconf.py

```diff
@@ -15,15 +17,34 @@ def file_format(_parent_: Container) -> str:
     return Path(_parent_["read_from"]).suffix
 
 
+def _get_architecture_capabilities(conf: BaseContainer) -> Dict[str, List[str]]:
+    architecture_name = conf["architecture"]["name"]
+    architecture = importlib.import_module(f"metatensor.models.{architecture_name}")
+    return architecture.__ARCHITECTURE_CAPABILITIES__
+
+
+def default_device(_root_: BaseContainer) -> str:
+    """Custom OmegaConf resolver to find the default precision of an architecture.
```

Comment:
Default device (not dtype), I also think this is not doing what the docstring says. This is picking the device for training, not finding the default device of the architecture